src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -907,6 +907,145 @@ public static async Task ZipArchive_InvalidHuffmanData()
+
+                updatedStream.Write(s_inconsistentVersionToExtract);
+                updatedStream.Seek(0, SeekOrigin.Begin);
+
+                using (ZipArchive originalArchive = new ZipArchive(updatedStream, ZipArchiveMode.Read, true))
+                {
+                    Assert.Equal(1, originalArchive.Entries.Count);
+
+                    ZipArchiveEntry firstEntry = originalArchive.Entries[0];
+
+                    Assert.Equal("first.bin", firstEntry.Name);
+                    Assert.Equal(10, firstEntry.Length);
+
+                    using (Stream entryStream = firstEntry.Open())
+                    {
+                        Assert.Equal(10, firstEntry.Length);
+
+                        byte[] uncompressedBytes = new byte[firstEntry.Length];
+                        int bytesRead = entryStream.Read(uncompressedBytes);
+
+                        Assert.Equal(10, bytesRead);
+
+                        Assert.Equal(new byte[] { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09 }, uncompressedBytes);
+                    }
+                }
+
+                updatedStream.Seek(0, SeekOrigin.Begin);
+
+                // Create a new entry, forcing the central directory headers to be rewritten. The local file header
+                // for first.bin would normally be skipped (because it hasn't changed) but it needs to be rewritten
+                // because the central directory headers will be rewritten with a valid value and the local file header
+                // needs to match.
+                using (ZipArchive updatedArchive = new ZipArchive(updatedStream, ZipArchiveMode.Update))
+                {
+                    ZipArchiveEntry newEntry = updatedArchive.CreateEntry("second.bin", CompressionLevel.NoCompression);
+                }
+
+                byte[] updatedContents = updatedStream.ToArray();
+                int updatedLocalVersionToExtract = updatedContents[4];
+                int updatedCentralDirectoryVersionToExtract = updatedContents[97];
+
+                Assert.Equal(20, updatedCentralDirectoryVersionToExtract);
+                Assert.Equal(20, updatedLocalVersionToExtract);
+            }
+        }
+
+        private static readonly byte[] s_inconsistentVersionToExtract =
+        {
+            // ===== Local file header signature 0x04034b50
+            0x50, 0x4b, 0x03, 0x04,
+            // version to extract 0.0 (invalid - this should be at least 2.0 to make use of deflate compression)
+            0x00, 0x00,
+            // general purpose flags
+            0x02, 0x00,   // 0000_0002 'for maximum-compression deflating'
+            // Deflate
+            0x08, 0x00,
+            // Last mod file time
+            0x3b, 0x33,
+            // Last mod date
+            0x3f, 0x5a,
+            // CRC32
+            0x46, 0xd7, 0x6c, 0x45,
+            // compressed size
+            0x0c, 0x00, 0x00, 0x00,
+            // UNcompressed size
```

Comment:
```suggestion
            // Uncompressed size
```

Reply:
Thanks - I've made the casing consistent here, and added an explanation to L923 of the test.